crates/evm/src/executor/backend/mod.rs

```diff
@@ -1079,37 +1081,48 @@ impl DatabaseExt for Backend {
                 // forks are selected `select_fork`
                 update_current_env_with_fork_env(env, fork_env);
 
-                // we also need to update the journaled_state right away, this has essentially the
-                // same effect as selecting (`select_fork`) by discarding
-                // non-persistent storage from the journaled_state. This which will
-                // reset cached state from the previous block
                 let mut persistent_addrs = self.inner.persistent_accounts.clone();
+
                 // we also want to copy the caller state here
                 persistent_addrs.extend(self.caller_address());
 
                 let active = self.inner.get_fork_mut(active_idx);
                 active.journaled_state = self.fork_init_journaled_state.clone();
 
+                // copy all the journaled states of persistent accounts
```

Comment:
I realize this is actually redundant in the master branch.. 

https://github.com/foundry-rs/foundry/blob/2885b0db02e620d3a2ce7a7e52650932005fa8ac/crates/evm/src/executor/backend/mod.rs#L1101-L1111

this would implicitly also cover the persistent accounts